server/clcachesrv.py

```diff
@@ -28,10 +32,11 @@ def getFileHash(self, path):
             hashsum = hashlib.md5(f.read()).hexdigest()
 
         watchedDirectory[basename] = hashsum
-        if dirname not in self._watchedDirectories:
+        if dirname not in self._watchedDirectories and not self.isExcluded(dirname) and not self._disableWatching:
```

Comment:
This condition makes me a bit dizzy, not only because of the number of boolean expressions involved but also because terms like `isExcluded` or `disableWatching` are negations in themselves. Not sure how to simplify it though.